### PR TITLE
docs(contributor): add governance document

### PR DESCRIPTION
**Pull request by kmova**
The top contributors for the OpenEBS projects are
from MayaData and Viking, and the current governance was primarily
done via a syncup on the community slack or with direct interactions.
As the project is gaining momentum and interest from more
companies to contribute, it is time to outline the
current governance model.

After looking at the various examples provided in https://github.com/openebs/openebs/issues/2545,
the closest that matches OpenEBS Governance mode is from:
https://github.com/containerd/project/blob/master/GOVERNANCE.md

The text in this document is borrowed from the above containerd
GOVERNANCE doc.

Please feel free to suggest changes and corrections.

Signed-off-by: kmova <kiran.mova@openebs.io>

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+Just contributing does not make you a maintainer, it is about building trust
+with the current maintainers of the project and being a person that they can
+depend on and trust to make decisions in the best interest of the project.
+
+Periodically, the existing maintainers curate a list of contributors that have
+shown regular activity on the project over the prior months. From this list,
+maintainer candidates are selected and proposed on the maintainers mailing list.
+
+After a candidate has been announced on the maintainers mailing list, the
```
Comment (gila) [inline GOVERNANCE.md]: Where are the mailing lists?

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+maintainer has already discussed the candidacy with the candidate and a
+maintainer is willing to be a sponsor by opening the pull request. The candidate
+becomes a maintainer once the pull request is merged.
+
+## Adding sub-projects
+
+Similar to adding maintainers, new sub projects can be added to OpenEBS
+GitHub organization as long as they adhere to the OpenEBS vision and mission. 
+New projects are discussed in either the Contributor Meeting or the Community 
+slack and requiers atleast 1 maintainer approval.
```
Comment (gila) [inline GOVERNANCE.md]: Don't think its a good idea to limit this to meetings or slack. Please consider adding mailing list as an option here as well

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+1) They share responsibility in the project's success.
+2) They have made a long-term, recurring time investment to improve the project.
+3) They spend that time doing whatever needs to be done, not necessarily what
+is the most interesting or fun.
+
+Maintainers are often under-appreciated, because their work is harder to appreciate.
+It's easy to appreciate a really cool and technically advanced feature. It's harder
+to appreciate the absence of bugs, the slow but steady improvement in stability,
+or the reliability of a release process. But those things distinguish a good
```
Comment (vishnuitta) [inline GOVERNANCE.md]: But, those things distinguish a great project from a good one.

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+## Adding sub-projects
+
+Similar to adding maintainers, new sub projects can be added to OpenEBS
+GitHub organization as long as they adhere to the OpenEBS vision and mission. 
+New projects are discussed in either the Contributor Meeting or the Community 
+slack and requiers atleast 1 maintainer approval.
+
+If a project is approved, a maintainer will add the project to the OpenEBS
+GitHub organization, and make an announcement on a public forum.
+
```
Comment (ksatchit) [inline GOVERNANCE.md]: @kmova , not sure if this is a correct question: On the same lines as above.. should we discuss the processes for addition/removal of features, roadmap items and other such as part of this doc OR is the scope of the governance doc strictly about repo maintenance with feature queries going into community docs.

File: GOVERNANCE.md (hunk 1 above)
Comment (richardelling) [reply GOVERNANCE.md]: How about "announced to the maintainers via email."

File: GOVERNANCE.md (hunk 3 above)
Comment (kmova) [reply GOVERNANCE.md]: updated.

File: GOVERNANCE.md (hunk 1 above)
Comment (kmova) [reply GOVERNANCE.md]: To keep this in sync with the current tools we already have setup, I have created a maintainers slack channel and updated the content accordingly. Once we get into CNCF sandbox, we could setup the mailing list as per the CNCF tools/guidelines and update the instructions about the mailing list usage.

File: GOVERNANCE.md (hunk 2 above)
Comment (kmova) [reply GOVERNANCE.md]: Will include the mailing list .. same response as above.

File: GOVERNANCE.md (hunk 4 above)
Comment (kmova) [reply GOVERNANCE.md]: @ksatchit - project management seems to be not included in the other sample governance docs I went through so far. But we could actually make it part of our doc by raising a PR to this doc. I have updated the contributor doc with details on the link to the community meetings page where we discuss about the features in the upcoming releases.

Comment (ksatchit) [reply GOVERNANCE.md]: Thanks @kmova

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+maintainer has already discussed the candidacy with the candidate and a
+maintainer is willing to be a sponsor by opening the pull request. The candidate
+becomes a maintainer once the pull request is merged.
+
+## Adding sub-projects
+
+Similar to adding maintainers, new sub projects can be added to OpenEBS
+GitHub organization as long as they adhere to the OpenEBS vision and mission. 
+New projects are discussed in either the Contributor Meeting or the Community 
+slack and requiers atleast 1 maintainer approval.
```
Comment (akhilerm) [inline GOVERNANCE.md]: @kmova , Typo  it should be `requires`

Comment (kmova) [reply GOVERNANCE.md]: thank you! fixed.

**Pull request merged**